HznApplication/HznApp.cpp

```diff
@@ -2,55 +2,234 @@
+{
+	std::string gridVertexShader = R"(
+		#version 420 core
+		layout (location = 0) in vec3 a_Pos;
+
+		uniform mat4 model;
+		uniform mat4 view;
+		uniform mat4 projection;
+		
+		void main()
+		{
+			gl_Position = projection * view * model * vec4(a_Pos, 1.0f);
+		}
+	)";
 
-//open & save file dependencies for win32 API
+	std::string gridFragmentShader = R"(
+		# version 420 core
+		
+		uniform vec4 a_Color;
+		out vec4 fragmentColor;
 
-#include <commdlg.h>
-#include "FileManagement/ProjectFile.h"
-#include "SceneManagement/Scene.h"
+		void main()
+		{
+			fragmentColor = a_Color;
+		}
+	)";
 
+	std::string textureVertexShader = R"(
+		#version 420 core
+		layout (location = 0) in vec3 a_Pos;
+		layout (location = 1) in vec2 a_TexCoord;
 
+		uniform mat4 model;
+		uniform mat4 view;
+		uniform mat4 projection;
 
+		out vec2 v_TexCoord;
+		
+		void main()
+		{
+			gl_Position = projection * view * model * vec4(a_Pos, 1.0f);
+			v_TexCoord = a_TexCoord;
+		}
+	)";
 
+	std::string textureFragmentShader = R"(
+		# version 420 core
+		
+		in vec2 v_TexCoord;
+		out vec4 fragmentColor;
 
+		uniform sampler2D f_Texture;
 
-std::shared_ptr<Hzn::App> Hzn::createApp()
-{
-	return std::make_shared<HznApp>();
-}
+		void main()
+		{
+			fragmentColor = texture(f_Texture, v_TexCoord);
+		}
+	)";
 
-// *********** SAMPLE LAYER **********
+	Hzn::BufferLayout layout =
+	{
+		{Hzn::ShaderDataType::Vec3f, "a_Pos"},
+		{Hzn::ShaderDataType::Vec2f, "a_TexCoord"}
+	};
+
+	std::vector<float> vertices = {
+		0.5f, 0.5f, 0.0f, 1.0f, 1.0f,
+		0.5f, -0.5f, 0.0f, 1.0f, 0.0f,
+		-0.5f, -0.5f, 0.0f, 0.0f, 0.0f,
+		-0.5f, 0.5f, 0.0f, 0.0f, 1.0f
+	};
+
+	std::vector<uint32_t> indices = {
+		0, 1, 2,
+		0, 3, 2
+	};
+
+	shader = std::shared_ptr<Hzn::Shader>(Hzn::Shader::create(gridVertexShader, gridFragmentShader));
+	textureShader = std::shared_ptr<Hzn::Shader>(Hzn::Shader::create(textureVertexShader, textureFragmentShader));
+
+	texture = Hzn::Texture2D::create("assets/textures/Checkerboard.png");
 
-SampleLayer::SampleLayer(const std::string& name) : Layer(name) {}
+	textureShader->bind();
+	textureShader->setUniform("f_Texture", 0);
+
+	vertexBuffer = std::shared_ptr<Hzn::VertexBuffer>(Hzn::VertexBuffer::create(vertices));
+	vertexBuffer->setBufferLayout(layout);
+
+	elementBuffer = std::shared_ptr<Hzn::ElementBuffer>(Hzn::ElementBuffer::create(indices));
+
+	vertexArray = std::shared_ptr<Hzn::VertexArray>(Hzn::VertexArray::create());
+	vertexArray->addVertexBuffer(vertexBuffer);
+	vertexArray->setElementBuffer(elementBuffer);
+
+	camera = std::shared_ptr<Hzn::Camera>(new Hzn::OrthographicCamera(
+		-1.0f, 1.0f, -1.0f, 1.0f, 0.0f, 100.0f,
+		{ 0.0f, 0.0f, 3.0f },
+		{ 0.0f, 0.0f, -1.0f },
+		{ 0.0f, 1.0f, 0.0f }
+	));
+
+	camera->speed = 2.5f;
+}
 
 void SampleLayer::onAttach()
 {
-	HZN_INFO("Sample Layer Attached!");
+	HZN_CRITICAL("Sample Layer Attached!");
 }
 
-void SampleLayer::onRenderImgui()
+void SampleLayer::onUpdate(Hzn::TimeStep deltaTime)
```

Comment:
I see we have implemented functions for TimeStep what are the uses of this and are there any other classes or parts where this would be utilised

Reply:
Timestep is a class the forms a wrapper around the float class, it holds time as a float value.